esp-hal-common/src/rsa/mod.rs

```diff
@@ -213,6 +346,42 @@ where
         self.rsa.clear_interrupt();
         Ok(())
     }
+
+    #[cfg(feature = "async")]
+    pub async fn read(&mut self, outbuf: &mut T::InputType) {
```

Comment:
We should remove this function, it's just a copy of `read_results` but its blocking.

`modular_multiplication` etc should call `block!(read_results(outbuf))` instead

Reply:
Maybe it'll be better to apply the same as we did with start_step2?
(I mean use loop idea instead of "nb::block!" and Err::WouldBlock)